src/transformers/generation_flax_utils.py

```diff
@@ -326,6 +326,14 @@ def generate(
         if decoder_start_token_id is None and self.config.is_encoder_decoder:
             raise ValueError("`decoder_start_token_id` has to be defined for encoder-decoder generation.")
 
+        # decoder-only models should use left-padding for generation (can't be checked with `trace=True`)
+        if not self.config.is_encoder_decoder and not trace:
```

Comment:
@patrickvonplaten -- without the `and not trace`, a `jax._src.errors.ConcretizationTypeError` is raised, because it's a boolean that depends on the contents of `input_ids`. This seemed the wisest path to inform the users without breaking the code

Also, FLAX is more picky than XLA TF, which doesn't complain about this check :D

Reply:
Sounds good!